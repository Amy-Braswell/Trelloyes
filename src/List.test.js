import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';

describe('Card Test Suite', () => {

    const STORE = {list: 
        {
        id: '1',
        header: 'First list',
        cardIds: [ 'a', 'b' ]
        },
        allCards: {
            'a': { id: 'a', title: 'First card', content: 'lorem ipsum' },
            'b': { id: 'b', title: 'Second card', content: 'lorem ipsum' }
        }
    };

    describe('Card Test Suite', () => {
        //Smoke Test
        it('renders without crashing', () => {
            // start by creating an element to render the component into
            const div = document.createElement('div');
            ReactDOM.render(<List 
                key={STORE.list.id}
                header={STORE.list.header}
                cards={STORE.list.cardIds}
                />,  
                div); 
            ReactDOM.unmountComponentAtNode(div);
        });

        //Snapshot Test
        it('renders the UI as expected', () => {
            const tree = renderer
                .create(<List 
                    key={STORE.list.id}
                    header={STORE.list.header}
                    cards={STORE.list.cardIds}
                />) 
                .toJSON();
            expect(tree).toMatchSnapshot();

        });
    });

});