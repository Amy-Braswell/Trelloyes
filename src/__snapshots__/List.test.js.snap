// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Test Suite Card Test Suite renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        delete
      </button>
      <h3 />
      <p />
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
